tools/rtpsrelay/RtpsRelay.cpp

```diff
@@ -936,6 +939,9 @@ int ACE_TMAIN(int argc, ACE_TCHAR* argv[])
   }
   ACE_DEBUG((LM_INFO, ACE_TEXT("(%P|%t) INFO: Meta Discovery listening on %C\n"), OpenDDS::DCPS::LogAddr(meta_discovery_addr).c_str()));
 
+  const bool has_run_time = config.run_time() != OpenDDS::DCPS::TimeDuration::zero_value;
```

Comment:
`TimeDuration` can be implicitly used as a `bool` to check if it's zero, so you could replace `has_run_time` with  `config.run_time()`.

Reply:
Or replace this line with:

```suggestion
  const bool has_run_time = !config.run_time().is_zero();
```